### PR TITLE
feat: add iron session cookies for auth handling

### PR DESCRIPTION
**Pull request by philipdesmedt**

### CODE REVIEW

**Comment by philipdesmedt** [comment]
@nieldeckx can you review pls?

this includes authentication setup (without message signing yet) and a basic DAO schema + UI form

File: web/common/fetchers.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: null,
+    });
+  } catch (e) {
+    console.log('ERROR:', e);
+  }
+};
+
+export const createUser = async (formData: Object) => {
```
Comment (nieldeckx) [inline web/common/fetchers.ts:23]: Q: were is this method used or why would it be needed?

File: web/common/constants.ts [hunk 2]
```diff
@@ -20,3 +20,6 @@ if (process.env.NETWORK === 'mainnet') {
   stacksNetwork = new StacksMainnet();
   stacksApiUrl = 'https://stacks-node-api.mainnet.stacks.co';
 } 
+
+const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
```
Comment (nieldeckx) [inline web/common/constants.ts:24]: PP: make all constants uppercase or lowercase instead

File: web/pages/api/dao/create.ts [hunk 3]
```diff
@@ -17,11 +17,14 @@ async function postHandler(
   res: NextApiResponse<Dao | string>
 ) {
   try {
+    console.log('Got request to create DAO', req.body);
     const prisma = new PrismaClient();
     const result = await prisma.dao.create({
       data: {
         publicKey: req.body.publicKey,
         title: req.body.title,
+        about: req.body.about,
```
Comment (nieldeckx) [inline web/pages/api/dao/create.ts:26]: REQ: update script `1-1-create-dao` to include new required info

File: web/pages/api/dao/create.ts [hunk 4]
```diff
@@ -17,11 +17,14 @@ async function postHandler(
...
+        amount: parseFloat(req.amount) * 100000000, // convert to sats
```
Comment (nieldeckx) [inline web/pages/api/dao/create.ts:27]: REQ: would use sats everywhere in backend/api to make it easy and not care about the notation. FE takes care of transforming user input in sats.

File: web/common/fetchers.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  await fetch(API_URL + '/api/user/create', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(formData),
+  });
+};
+
+export const createDao = async (formData: Object) => {
```
Comment (nieldeckx) [inline web/common/fetchers.ts:33]: PP: change filename as this method will post data :P

File: web/common/session-helpers.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as Iron from 'iron-session';
```
Comment (nieldeckx) [inline web/common/session-helpers.ts:1]: PP: put session files in separate directory

**Review by nieldeckx — CHANGES_REQUESTED** [review]
Nice work. Some comments to keep codebase cleaner imo.

File: web/common/fetchers.ts (hunk 1 above)
Comment (philipdesmedt) [reply web/common/fetchers.ts:23]: Didn't need it (yet). I can remove it until we need it again

File: web/pages/api/dao/create.ts (hunk 3 above)
Comment (philipdesmedt) [reply web/pages/api/dao/create.ts:26]: ok

File: web/pages/api/dao/create.ts (hunk 4 above)
Comment (philipdesmedt) [reply web/pages/api/dao/create.ts:27]: Yes - everything in back-end is sats, but this conversion is on the back-end and comes in in BTC from the front-end so we need to convert it

File: web/common/fetchers.ts (hunk 5 above)
Comment (philipdesmedt) [reply web/common/fetchers.ts:23]: well, the `fetchers` refer to the `fetch` library which does both POST and GET calls, but I can rename it if you want :D

**Comment by philipdesmedt** [comment]
@nieldeckx incorporated all feedback except for the fetchers (see comment). Gonna merge this and continue on the DAO creation + funding part

**Pull request merged**